test/parallel/test-crypto-subtle-cross-realm.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const { subtle } = crypto;
+const vm = require('vm');
+
+// Test with same-realm ArrayBuffer
+{
+  const samerealmData = new Uint8Array([1, 2, 3, 4]).buffer;
+
+  subtle.digest('SHA-256', samerealmData)
+    .then((result) => {
+      assert(result instanceof ArrayBuffer);
+      assert.strictEqual(result.byteLength, 32); // SHA-256 is 32 bytes
+    })
```

Comment:
```suggestion
    .then(common.mustCall((result) => {
      assert(result instanceof ArrayBuffer);
      assert.strictEqual(result.byteLength, 32); // SHA-256 is 32 bytes
    }))
```

And instead of adding the `.catch(common.mustNotCall())`, use `assert.rejects(...)` otherwise if the asserts within the `then(...)` fail the errors will be swallowed by the error indicating the `.catch(..)` handler was invoked.

Reply:
The reason the catch handler here isn't helpful is that if the asserts in the then handler fail, those errors will be squashed by the error thrown when the catch handler is invoked and the test log won't see the actual assert that failed.